### PR TITLE
feat: order creation completed

### PR DESCRIPTION
**Pull request by andreoliveiramorais**
## Description
[Notion card](https://www.notion.so/)

Please include a summary of the changes introduced by this PR. Please also include relevant motivation and context.

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. If possible, please provide screenshots,
curls or logs to help explain the changes.

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: domain/src/main/java/postech/soat/tech/challenge/model/order/Order.java [hunk 1]
```diff
@@ -49,6 +50,11 @@ private void validate() {
             domainValidationResult.addError("An order must have at least one combo");
             throw new DomainInvalidException(domainValidationResult.getErrors(), domainValidationResult.getErrorsMessage());
         }
+        if (customer == null) {
+            var domainValidationResult = new DomainValidationResult();
+            domainValidationResult.addError("An order must have a customer");
```
Comment (gspaiva) [inline domain/src/main/java/postech/soat/tech/challenge/model/order/Order.java:55]: E quando o customer não se cadastrar?

File: infrastructure/src/main/java/postech/soat/tech/challenge/api/request/order/CreateOrderRequestDTO.java [hunk 2]
```diff
@@ -11,6 +11,7 @@
 @Getter
 @NoArgsConstructor
 public class CreateOrderRequestDTO {
+    private long customerId;
```
Comment (gspaiva) [inline infrastructure/src/main/java/postech/soat/tech/challenge/api/request/order/CreateOrderRequestDTO.java:14]: ditto

**Review by r4ulmanzas — APPROVED** [review]

File: infrastructure/src/main/java/postech/soat/tech/challenge/persistence/entity/order/OrderEntity.java [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+    @Id
+    @GeneratedValue(strategy = GenerationType.AUTO, generator = "customer_order_id_seq")
+    @SequenceGenerator(name = "customer_order_id_seq", sequenceName = "customer_order_id_seq", allocationSize = 1)
+    private Long id;
+    @OneToMany(mappedBy = "order", cascade = CascadeType.ALL, orphanRemoval = true)
+    private List<ComboEntity> combos;
+    @Column(name = "order_value")
+    private BigDecimal value;
+    @OneToOne
+    @JoinColumn(name = "customer_id", nullable = false)
```
Comment (gspaiva) [inline infrastructure/src/main/java/postech/soat/tech/challenge/persistence/entity/order/OrderEntity.java:33]: Aqui poderia ser nullable pq não é obrigatório ter um customer, ah não ser que vamos gerar um fake customer para os casos o cliente não quiser se identificar.

File: infrastructure/src/main/resources/db/migration/V6__create_order_table.sql [hunk 4]
```diff
@@ -0,0 +1,10 @@
+create table customer_order (
```
Comment (gspaiva) [inline infrastructure/src/main/resources/db/migration/V6__create_order_table.sql:1]: Mesma coisa aqui, se considerarmos que o cliente não é obrigatório, não deveriamos chamar de customer_order, somente order

File: infrastructure/src/main/resources/db/migration/V6__create_order_table.sql [hunk 5]
```diff
@@ -0,0 +1,10 @@
...
+    id bigserial primary key,
+    order_value numeric(10, 2) not null,
+    customer_id bigserial not null,
```
Comment (gspaiva) [inline infrastructure/src/main/resources/db/migration/V6__create_order_table.sql:4]: ditto

File: infrastructure/src/main/java/postech/soat/tech/challenge/persistence/entity/CustomerEntity.java [hunk 6]
```diff
@@ -24,9 +25,11 @@ public class CustomerEntity {
     private String email;
     @Column
     private String phone;
+    @OneToOne(mappedBy = "customer", cascade = CascadeType.ALL, orphanRemoval = true)
```
Comment (gspaiva) [inline infrastructure/src/main/java/postech/soat/tech/challenge/persistence/entity/CustomerEntity.java:28]: Teria que ser um relacionamento Many to Many caso vejamos necessidade para essa primeira entrega.

**Review by gspaiva — APPROVED** [review]
Aprovando mas com ressalvas.

File: domain/src/main/java/postech/soat/tech/challenge/model/order/Order.java (hunk 1 above)
Comment (felipetomm) [reply domain/src/main/java/postech/soat/tech/challenge/model/order/Order.java:55]: Cadastrei um customer inicial nas migrations pra garantir que teremos um default. Achei melhor contornar assim pra mim conseguir seguir nos demais pontos.

**Review by felipetomm — APPROVED** [review]

**Pull request merged**